bitcoin/src/blockdata/transaction.rs

```diff
@@ -835,6 +835,50 @@ impl Transaction {
     }
 }
 
+/// The transaction version.
+///
+/// Currently, as specified by [BIP-68], only version 1 and 2 are considered standard.
+///
+/// Standardness of the inner `i32` is not an invariant because you are free to create transactions
+/// of any version, transactions with non-standard version numbers will not be relayed by the
+/// Bitcoin network.
+///
+/// [BIP-68]: https://github.com/bitcoin/bips/blob/master/bip-0068.mediawiki
+#[derive(Copy, PartialEq, Eq, Clone, Debug, PartialOrd, Ord, Hash)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+#[cfg_attr(feature = "serde", serde(crate = "actual_serde"))]
+pub struct Version(pub i32);
+
+impl Version {
+    /// The original Bitcoin transaction version (pre-BIP-68).
+    pub const ONE: Self = Self(1);
+
+    /// The second Bitcoin transaction version (post-BIP-68).
+    pub const TWO: Self = Self(2);
+
+    /// Creates a non-standard transaction version.
+    pub fn non_standard(version: i32) -> Version { Self(version) }
+
+    /// Returns true if this transaction version number is considered standard.
+    pub fn is_standard(&self) -> bool { *self == Version::ONE || *self == Version::TWO }
+}
+
+impl Default for Version {
```

Comment:
post merge comment:

1) Is this a reasonable default. We are expecting transaction version 3 sometime in the future. If we silently change this to three, it will break a lot of code downstream. 

I don't think this should implement Default as there is no sane default that we can represent. If we want a default I think it should be zero.

Reply:
@tcharding, @apoelstra, I feel this needs to addressed/discussed.

Reply:
If version 3 transactions ship and the default version changes, then we would simply release a new breaking version of the library with release notes.

I think a sane default would be what Core uses. Version zero would be nonstandard, right?

Reply:
Some interesting points, I could easily have changed a default value and not thought it a breaking change. I don't see a lot of negative in removing the `Default` so if there is the slightest concern I think we should do that.

Reply:
I didn't realize that version 3 was coming down the pike. I agree, we should drop the default (though I'd like to put it back as 3 whenever 3 comes out, because that default will last for several years at least).

@clarkmoody it is a breaking change so in principle we're justified in messing with the value in any breaking release, but I think changing this value would be extremely hard to notice -- if we did this to Liquid I'd bet we'd go literally years before we noticed that we were producing transactions with a different version number, and then we'd need some forensics to see when it happened (and we'd only even notice because we happen to know that Liquid long predates tx v3 so it can't have *always* been 3..). Because there's no "visible" effect other than a potential change in privacy set.

In the case of Liquid it wouldn't matter at all, but if we were a user who cared for some reason, we'd be upset.